lib/board.rb

```diff
@@ -0,0 +1,118 @@
+class Board
+  attr_reader :cells
+
+  def initialize
+    @cells = [
+      [1, 2, 3],
+      [4, 5, 6],
+      [7, 8, 9]
+    ]
+
+    @moves = 0
+  end
+
+  def cell_available?(number)
+    false if number < 1 || number > 9
+    @cells.each do |row|
+      row.each { |cell| return true if cell == number }
+    end
+    false
+  end
+
+  def draw
+    @cells
+  end
+
+  def move(number, character)
+    return -2 unless moves?
+    return -1 unless cell_available?(number)
+
+    @cells.each do |row|
+      row.each_index do |index|
+        next unless row[index] == number
+
+        row[index] = character
+        @moves += 1
+        if @moves > 2
+          return 1 if check_win?(character)
+        end
+      end
+    end
+    0
+  end
+
+  def moves?
+    return true if @moves <= 9
+
+    false
+  end
+
+  def check_win_rows?(character)
+    @cells.each do |row|
+      count = 0
+      row.each do |cell|
+        if cell == character
+          count += 1
+          return true if count == 3
+        elsif count
+          count -= 1
+        end
+      end
+    end
+    false
+  end
+
+  def check_win_columns?(character)
+    (0..2).each do |j|
+      count = 0
+      (0..2).each do |i|
+        if @cells[i][j] == character
+          count += 1
+          return true if count == 3
+        elsif count
+          count -= 1
+        end
+      end
+    end
+    false
+  end
```

Comment:
**Optional** an easier way to check for the winner is to have all possible combinations and check the array against them, using the player's character as an argument.